exp/query/lsm_fetcher.go

```diff
@@ -123,20 +152,13 @@ func (l *LSMFetcher) Fetch(ctx context.Context, collection, tenant string, versi
 			// NOTE: Removing old version can be tricky, even though all new lookup returns new version path.
 			// There can be some clients still using old versions. We can leave without removing and delete only if eviction is needed.
 			// But still that doesn't remove above stated problem.
-			if l.remove(ctx, basePath, collection, tenant, oldVersion); err != nil {
-				return nil, "", err
+			if err := l.remove(basePath, collection, tenant, oldVersion); err != nil {
+				return "", fmt.Errorf("failed to remove old version of tenant data: %w", err)
```

Comment:
NIT: can we add here to the error message which log and which version ?

RAtionale: if this keep happening to the same tenant+version ops can manually delete it if it notices it

Reply:
Hmm. good question. I have an opinion here.

Any error should be "handled" once. Handling here means, you see the error and make some decision. This can be either returning to the caller or logging it in the same function. But never both. I will explain why.

1. "why not both return and log error" - Because, both returning the error and logging it almost always end up causing confusion, because the caller can log as well or do something else (passing all the way up to the stack and they themselfs can also log it). This will lead to cluttering logs with redundant error messages and make it hard to correlate events in logs during debugging or investigating.

3. "why I don't want to log here and instead just like to return the error" - Because I want our querier to fail early if anything wrong with one of it's core dependency, the disk. I want to return 5xx when something wrong with disk and like to do it as early as possible. If we just log it and not return error, then it may serve the requests (successfully) for a while before the problem build up and fail in a weird ways that may be hard to debug (even to narrow down when the disk started failing)

Having said all that, nothing written on stone. once we take it prod, we can learn and change this remove failures differently may be.